### PR TITLE
implement and test ThingIFAPI#getTargetState.

### PR DESCRIPTION
**Pull request by naoki-akimoto**
ThingIFAPI#getTargetState 2つの実装を行いテストを追加しました。レビューをお願いします。

TargetStateのサブクラスのAdapterは要らない模様です。

### CODE REVIEW

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 1]
```diff
@@ -1548,8 +1549,30 @@ public String deleteTrigger(
...
+                this.app.getAppID(), this.target.getTypedID().toString());
+        String url = Path.combine(this.app.getBaseUrl(), path);
+        Map<String, String> headers = this.newHeader();
+        IoTRestRequest request = new IoTRestRequest(url, IoTRestRequest.Method.GET, headers);
+
+        JSONObject responseBody = this.restClient.sendRequest(request);
+        Map retMap = new HashMap<>();
+        Iterator it = responseBody.keys();
+        while (it.hasNext()) {
+            String key = (String)it.next();
```
Comment (yongpingchen) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: check whether the key is in `this.stateTypes` or not? 
```
if (!this.stateTypes.has(key)) {
    throw new UnregisteredAliasException(key, false);
}
```

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 2]
```diff
@@ -1548,8 +1549,30 @@ public String deleteTrigger(
...
+            try {
+                JsonObject object = new JsonParser()
+                        .parse(responseBody.getJSONObject(key).toString()).getAsJsonObject();
+                retMap.put(key, this.gson.fromJson(object, this.stateTypes.get(key)));
+            } catch (JSONException e) {
+                throw new IllegalStateException("Unknown target state class: " + key);
```
Comment (yongpingchen) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: not throw exception here. Better to check whether key is in this.stateTypes before.

**Comment by shinsuke-nara** [comment]
@naoki-akimoto LGTM.

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 3]
```diff
@@ -1569,19 +1592,19 @@ public String deleteTrigger(
...
-//        String path = MessageFormat.format("/thing-if/apps/{0}/targets/{1}/states", this.app.getAppID(), this.target.getTypedID().toString());
-//        String url = Path.combine(this.app.getBaseUrl(), path);
-//        Map<String, String> headers = this.newHeader();
-//        IoTRestRequest request = new IoTRestRequest(url, IoTRestRequest.Method.GET, headers);
-//        JSONObject responseBody = this.restClient.sendRequest(request);
-//        S ret = GsonRepository.gson().fromJson(responseBody.toString(), classOfS);
-//        return ret;
-        return null;
+        if (!this.stateTypes.containsKey(alias)) {
+            throw new IllegalArgumentException("Unknown alias: " + alias);
```
Comment (yongpingchen) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: throw `UnregisteredAliasException `

**Comment by naoki-akimoto** [comment]
@yongpingchen fixed. Please confirm.

**Comment by naoki-akimoto** [comment]
@yongpingchen added. Please confirm.

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 4]
```diff
@@ -1586,6 +1587,7 @@ public String deleteTrigger(
      * @throws ThingIFException Thrown when failed to connect IoT Cloud Server.
      * @throws ThingIFRestException Thrown when server returns error response.
      * @throws ClassCastException Thrown when S is not registered.
```
Comment (yongpingchen) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: > @throws ClassCastException Thrown when S is not registered.

It is better to improve this Exception 
```
@throws ClassCastException Thrown when S is different with registered class for this alias. 
```

**Pull request merged**